river/Root.zig

```diff
@@ -260,6 +260,11 @@ pub fn startTransaction(self: *Self) void {
     // Iterate over all views of all outputs
     var output_it = self.outputs.first;
     while (output_it) |output_node| : (output_it = output_node.next) {
+        // If a layout demand is currently in progress for this output,
+        // skip transaction on this output to avoid imperfect frames.
+        // We take Waylands motto, "every frame is perfect", seriously.
+        if (output_node.data.layout_demand != null) continue;
```

Comment:
This should preempt the current layout demand and make a new one

Reply:
why?